bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/clangd/ClangdConfigurationFileManager.java

```diff
@@ -90,11 +94,12 @@ protected void setCompilationDatabasePath(IProject project, ICProjectDescription
 				if (!relativeDatabasePath.isEmpty()) {
 					try {
 						setCompilationDatabase(project, relativeDatabasePath);
-					} catch (ScannerException e) {
+					} catch (Exception e) {
+						// catch all exceptions:
 						var status = new Status(IStatus.ERROR, ClangdPlugin.PLUGIN_ID, e.getMessage());
 						var projectLocation = project.getLocation().addTrailingSeparator().toPortableString();
-						LspUtils.showErrorMessage(LspEditorUiMessages.CProjectChangeMonitor_yaml_scanner_error,
-								LspEditorUiMessages.CProjectChangeMonitor_yaml_scanner_error_message + projectLocation
+						showErrorDialog(LspEditorUiMessages.ClangdConfigurationFileManager_yaml_error,
```

Comment:
Since this is a component, you can try to bind `org.eclipse.e4.core.services.statusreporter.StatusReporter` and report through it.

Reply:
> But it should be a headless component implementation. Why does it show any dialogs at all?

Users can edit the the .clangd file manually to add more options. If they add failures, it will be revealed when the .clangd file gets updated.

Reply:
> Since this is a component, you can try to bind `org.eclipse.e4.core.services.statusreporter.StatusReporter` and report through it.

yes. I'll cehck that.

Reply:
> Users can edit the the .clangd file manually to add more options. If they add failures, it will be revealed when the .clangd file gets updated.

Maybe it's better to add a listener when the file gets edited and validate it (e.g. by loading it and showing an error dialog if the load fails).

Reply:
> Users can edit the the .clangd file manually to add more options. If they add failures, it will be revealed when the .clangd file gets updated.

But this is different story. If user initiated all the chain from UI, we need to throw an exception from this method and show it to the user in UI layer. In this case you don't need more services, just throw